app/src/service/ocr/parser.ts

```diff
@@ -0,0 +1,78 @@
+import { logger as ocrLogger } from ".";
+
+export type EmployerData = {
+  employerIdentificationNumber: string;
+  wagesTipsOthers: string;
+  federalIncomeTaxWithheld: string;
+};
+
+export type ParsedData = {
+  employer: EmployerData;
+  employeeAddress: string;
+  ssn: string;
+  bottomLines: string;
+};
+
+export const parseEarningsStatement = (
+  text: string,
+  logger: typeof ocrLogger
+): Partial<EmployerData> => {
+  logger.info("Parsing earnings statement");
+
+  const results = {} as Partial<EmployerData>;
+
+  // a variable (a map) to store the regex and field name to be used in the loop
+  const patterns = {
+    company: /(H GREG NISSAN DELRAY LLC)/,
```

Comment:
We'll want to come up with some different parsing logic to extract company amongst other fields `(employeeName, employeeAddress, etc.)` which is completely doable. The most important information (earnings and pay date) are captured.